pytest_odoo.py

```diff
@@ -110,6 +111,30 @@ def load_http(request):
         signal.signal(signal.SIGINT, signal.default_int_handler)
 
 
+@contextmanager
+def _worker_db_name():
+    # This method ensure that if tests are ran in a distributed way
+    # thanks to the use of pytest-xdist addon, each worker will use
+    # a specific copy of the initial database to run their tests.
+    # In this way we prevent deadlock errors.
+    xdist_worker = os.getenv("PYTEST_XDIST_WORKER")
+    original_db_name = db_name = odoo.tests.common.get_db_name()
+    try:
+        if xdist_worker:
+            db_name = f"{original_db_name}-{xdist_worker}"
+            ret = os.system(f"psql -lqt | cut -d \| -f 1 | grep -w {db_name}")
+            if ret == 0:
+                os.system(f"dropdb {db_name}")
```

Comment:
Wouldn't it be simpler like this?

```suggestion
            os.system(f"dropdb {db_name} --if-exists")
```